File: 2023/1/C#/1.cs
using System.Text.RegularExpressions;

string[] input = File.ReadAllLines("./../../../../input.txt");
int[] sumList;

input = input.Select(s => s
	.Replace("one", "o1e")
    .Replace("two", "t2o")
    .Replace("three", "t3e")
    .Replace("four", "f4")
    .Replace("five", "f5e")
    .Replace("six", "s6")
    .Replace("seven", "s7n")
    .Replace("eight", "e8t")
    .Replace("nine", "n9")).ToArray();

for (int i = 0; i < input.Length; i++) {
    MatchCollection numbersInString = new Regex(@"\b[0-9]\w*\b").Matches(input[i]);

    foreach (var number in numbersInString)
    {
        Console.WriteLine(number);
    }
}


Console.ReadLine();

File: 2023/1/1.js
const { readLines, log, singleDigits, sum } = require("./../common")

let input = readLines("./input.txt")

input = input.map(line => line
	.replace(/one/g, "o1e")
	.replace(/two/g, "t2o")
	.replace(/three/g, "t3e")
	.replace(/four/g, "f4")
	.replace(/five/g, "f5e")
	.replace(/six/g, "s6")
	.replace(/seven/g, "s7n")
	.replace(/eight/g, "e8t")
	.replace(/nine/g, "n9e")
)

const sumList = []

for (let i = 0; i < input.length; i++) {
	const numbersInString = singleDigits(input[i])

	if (numbersInString) {
		const stringToInt = numbersInString.map(match => parseInt(match, 10))

		if (stringToInt.length === 1) {
			const concatenatedNumbers = parseInt(stringToInt[0].toString() + stringToInt[0].toString(), 10)

			sumList.push(concatenatedNumbers)
		} else {
			const firstNumber = stringToInt[0]
			const lastNumber = stringToInt[stringToInt.length - 1]
			const concatenatedNumbers = parseInt(firstNumber.toString() + lastNumber.toString(), 10)

			sumList.push(concatenatedNumbers)
		}
	}
}

log(sum(sumList))